playwright/_impl/_browser_type.py

```diff
@@ -187,6 +189,9 @@ async def connect(
         if timeout is None:
             timeout = 30000
 
+        if headers is None:
+            headers = {"User-Agent": getUserAgent()}
```

Comment:
are you sure that you only want to add it when there are no headers given? I would do it on another layer - on the transport directly.